TotalCrossSDK/src/main/java/totalcross/json/JSONFactory.java

```diff
@@ -110,95 +119,112 @@ public void setDescription(String description)
-        if (jsonObject.has(originalName)) {
-          name = originalName;
-        } else if (!jsonObject.has(name = originalName.toLowerCase())) {
-          // not found as-is or lowercased? try replacing camel case with underscore
-            /*
-             * originally done using regex, but totalcross implementation has some bugs and
-             * until they are fixed this is done looping through the characters
-             * originalName.replaceAll("(.)(\\p{Upper})", "$1_$2").toLowerCase();
-             */
-          StringBuilder sb = new StringBuilder();
-          boolean lastWasUnderscored = false;
-          for(int i = 0 ; i < originalName.length() ; i++) {
-              char c = originalName.charAt(i);
-              if (!lastWasUnderscored && Character.isUpperCase(c)) {
-                  lastWasUnderscored = true;
-                  sb.append('_');
-              } else {
-                  lastWasUnderscored = false;
-              }
-              sb.append(Character.toLowerCase(c));
-          }
-          final String underscoredName = sb.toString();
-          if (jsonObject.has(underscoredName)) {
-            name = underscoredName;
-          }
-        }
-        if (!jsonObject.isNull(name)) {
-          Class<?> parameterType = method.getParameterTypes()[0];
-          if (parameterType.isPrimitive()) {
-            if (parameterType.isAssignableFrom(boolean.class)) {
-              method.invoke(object, jsonObject.getBoolean(name));
-            } else if (parameterType.isAssignableFrom(int.class)) {
-              method.invoke(object, jsonObject.getInt(name));
-            } else if (parameterType.isAssignableFrom(long.class)) {
-              method.invoke(object, jsonObject.getLong(name));
-            } else if (parameterType.isAssignableFrom(double.class)) {
-              method.invoke(object, jsonObject.getDouble(name));
-            }
-          } else if (parameterType.isAssignableFrom(String.class)) {
-            method.invoke(object, jsonObject.getString(name));
-          } else if (parameterType.isAssignableFrom(Double.class)) {
-            method.invoke(object, jsonObject.getDouble(name));
-          } else if (parameterType.isAssignableFrom(Integer.class)) {
-            method.invoke(object, jsonObject.getInt(name));
-          } else if (parameterType.isAssignableFrom(Long.class)) {
-            method.invoke(object, jsonObject.getLong(name));
-          } else if (parameterType.isAssignableFrom(Boolean.class)) {
-            method.invoke(object, jsonObject.getBoolean(name));
-          } else if (parameterType.isArray()) {
-            method.invoke(object, parse(object, jsonObject.getJSONArray(name), parameterType));
+        // name as-is
+        methodCache.put(originalName, method);
+        // lowercased name
+        methodCache.put(originalName.toLowerCase(), method);
+        // not found as-is or lowercased? try replacing camel case with underscore
+        /*
```

Comment:
I think that it is safe to assume that a private method will be inlined by the javac compiler with proper given optimization, so for code clarity I think that would be better it this snippet that gives snakecase names was removed to a proper method

Also, I think that `getABCD` will yield as a result `a_bc_d`. I have no idea if it is intended to behave like this, but I find it odd and surprises me.